pkg/beacon/relay/event/deduplicator.go

```diff
@@ -0,0 +1,132 @@
+package event
+
+import (
+	"encoding/hex"
+	"fmt"
+	"sync"
+)
+
+// Local chain interface to avoid import cycles.
+type chain interface {
+	CurrentRequestPreviousEntry() ([]byte, error)
+}
+
+// Deduplicator decides whether the given event should be handled by the
+// client or not.
+//
+// Event subscription may emit the same event two or more times. The same event
+// can be emitted right after it's been emitted for the first time. The same
+// event can also be emitted a long time after it's been emitted for the first
+// time. It is deduplicator's responsibility to decide whether the given
+// event is a duplicate and should be ignored or if it is not a duplicate and
+// should be handled.
+//
+// Those events are supported:
+// - group selection started
+// - relay entry requested
+type Deduplicator struct {
+	chain                           chain
+	minGroupSelectionDurationBlocks uint64
```

Comment:
`minGroupCreationDurationBlock`? For the same reason as explained earlier.

Reply:
https://github.com/keep-network/keep-core/pull/2574/commits/7423b139421edc721f4cf33b75d62509fa98dc1a